### PR TITLE
[freeboxos] Complete Alarm system handling

### LINKED ISSUE #17220: [freeboxos] Add PIR management capability
The PIR is currently not handled in the Home capabilities.

### PR DESCRIPTION
**Pull request by clinique**
Resolves #17220 
Adds more states to Alarm, Keyfob
Corrects basic-shutter not being created.

Home requests can be quite large, so now the binding uses its own httpClient with resized buffer response.

### CODE REVIEW

**Comment by clinique** [comment]
@ben12 : FYI

File: bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/api/ApiHandler.java [hunk 1]
```diff
@@ -86,6 +88,21 @@ public ApiHandler(HttpClient httpClient, TimeZoneProvider timeZoneProvider) {
...
+        } catch (Exception e) {
+            throw new IllegalArgumentException("Unable to start httpClient", e);
+        }
+    }
+
+    public void dispose() {
+        try {
+            httpClient.stop();
+        } catch (Exception e) {
+            throw new IllegalArgumentException("Unable to stop httpClient", e);
```
Comment (lolodomo) [inline bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/api/ApiHandler.java:104]: A simple DEBUG log would be sufficient.

**Review by lolodomo — COMMENTED** [review]
Review part 1 of 2

File: bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/api/ApiHandler.java (hunk 1 above)
Comment (clinique) [reply bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/api/ApiHandler.java:104]: I guess this is more than unlikely to happen but when it happens I suppose it is suitable to have something more visible than a DEBUG log, no ?

Comment (jlaur) [reply bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/api/ApiHandler.java:104]: When looking at other bindings, I see both **warn** and **debug** logging used as well as fully ignoring it. So logging a warning wouldn't be as controversial as throwing an `IllegalArgumentException`. 😉

**Comment by lolodomo** [comment]
@clinique : you have a conflict to resolve after the merge of your other PR.

**Comment by clinique** [comment]
> @clinique : you have a conflict to resolve after the merge of your other PR.

That's what I'm currently doing :-)

**Review by lolodomo — CHANGES_REQUESTED** [review]

**Comment by lolodomo** [comment]
And sorry but you will have to rebase again.

File: bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/channel-types.xml [hunk 2]
```diff
@@ -503,4 +503,45 @@
...
+	<channel-type id="kfb-pushed">
+		<item-type>Number</item-type>
+		<label>Key Code Pushed</label>
+		<description>Last key pushed on the remote</description>
+		<state readOnly="true" pattern="%d">
+			<options>
+				<option value="1">Arm Absent Mode</option>
+				<option value="2">Disarm</option>
+				<option value="3">Arm Night Mode</option>
+				<option value="4">Point</option>
```
[the remark this answers is not in the document]
Comment (clinique) [reply bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/channel-types.xml:527]: I did not know how to name it : 
image

But BTW this button is not used so I could remove it.

File: bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/home-thing-type.xml [hunk 3]
```diff
@@ -21,6 +21,33 @@
...
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="api"/>
+		</supported-bridge-type-refs>
+
+		<label>Freebox Home PIR</label>
+		<description>A motion Sensor</description>
+
+		<channels>
+			<channel id="trigger" typeId="pir-trigger"/>
+			<channel id="trigger-update" typeId="timestamp">
```
[the remark this answers is not in the document]
Comment (clinique) [reply bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/home-thing-type.xml:34]: You're correct !

File: bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/channel-types.xml [hunk 4]
```diff
@@ -503,4 +503,45 @@
...
 
+	<channel-type id="pir-trigger">
+		<item-type>Switch</item-type>
+		<label>Détection</label>
+		<category>oh:freeboxos:mouvement</category>
+	</channel-type>
+
+	<channel-type id="pir-cover">
+		<item-type>Contact</item-type>
+		<label>Couvercle</label>
```
[the remark this answers is not in the document]
Comment (clinique) [reply bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/channel-types.xml:514]: You're right, this is the "Cover Tamper"

[the remark this answers is not in the document]
Comment (lolodomo) [reply bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/channel-types.xml:514]: Ok, it was what I guess.

File: bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/home-thing-type.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (lolodomo) [reply bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/home-thing-type.xml:34]: In that case, of course update also the other channel types.

File: bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/channel-types.xml (hunk 4 above)
[the remark this answers is not in the document]
Comment (clinique) [reply bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/channel-types.xml:514]: Changed

File: bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/channel-types.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (clinique) [reply bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/channel-types.xml:527]: Removed

File: bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/channel-types.xml [hunk 5]
```diff
@@ -503,4 +503,45 @@
...
+				<option value="2">Disarm</option>
+				<option value="3">Arm Night Mode</option>
+				<option value="4">Point</option>
+			</options>
+		</state>
+	</channel-type>
+
+	<channel-type id="alarm-state">
+		<item-type>String</item-type>
+		<label>Alarm state</label>
```
[the remark this answers is not in the document]
Comment (clinique) [reply bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/channel-types.xml:534]: Corrected

File: bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/home-thing-type.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (clinique) [reply bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/home-thing-type.xml:34]: Modified

**Review by lolodomo — CHANGES_REQUESTED** [review]

**Pull request merged**